File: esbmc.cpp/validador_preco/validador_caracteres_valor.cpp
#include <cctype>

#define MAX_PRICE_LEN 6

using namespace std;

bool validate_characters(const char price[MAX_PRICE_LEN], int len) {
    int dot_count = 0;
    for (int i = 0; i < len; i++) {
        char c = price[i];
        if (c == '.') {
            dot_count++;
        } else {
            __ESBMC_assert(isdigit(c), "O preço deve conter apenas números e, no máximo, um ponto.");
        }
    }
    __ESBMC_assert(dot_count <= 1, "O preço não pode conter mais de um ponto.");
    return true;
}

int main() {
    char price[MAX_PRICE_LEN];
    int len;

    __ESBMC_assume(len >= 1 && len <= 5);

    validate_characters(price, len);

    return 0;
}

File: esbmc.py/Testes02/validador_preco/validador_caracteres_valor.py
MAX_PRICE_LEN: int = 6

def validate_characters(price: str, length: int) -> bool:
    dot_count: int = 0
    i: int = 0
    while i < length:
        c: str = price[i]
        if c == '.':
            dot_count += 1
        else:
            assert c.isdigit(), "O preço deve conter apenas números e, no máximo, um ponto."
        i += 1
    assert dot_count <= 1, "O preço não pode conter mais de um ponto."
    return True


def main() -> None:
    price: str = input()          
    length: int = len(price)

    assert length >= 1 and length <= 5

    validate_characters(price, length)


main()
